R/pagination.R

```diff
@@ -256,13 +276,20 @@ valid_pag <- function(pagdf,
     reflines <- reflines + div_height + 1L
   }
 
-  ##  reflines <- sum(pagdf[start:guess, "nreflines"])
+
   rowlines <- raw_rowlines + reflines ## sum(pagdf[start:guess, "self_extent"]) - reflines
   ## self extent includes reflines
-  ## self extent does ***not*** currently include trailing sep
+  ## self extent does ***not*** currently include trailing sep for rows
+  ## self extent does ***not*** currently include col_gap for columns
   ## we don't include the trailing_sep for guess because if we paginate here it won't be printed
-  sectlines <- if (start == guess) 0L else sum(!is.na(pagdf[start:(guess - 1), "trailing_sep"]))
-  lines <- rowlines + sectlines # guess - start + 1 because inclusive of start
+  ncols <- 0L
+  if (row) {
+    sectlines <- if (start == guess) 0L else sum(!is.na(pagdf[start:(guess - 1), "trailing_sep"]))
+  } else { ## columns
+    ncols <- guess - start + 1 + length(pagdf$reprint_inds[[start]]) ## +1 because its incluive, 5-6 is 2 columns
+    sectlines <- col_gap * (ncols - as.integer(!has_rowlabels)) ## -1 if no row labels
```

Comment:
this is part of the col_gap fix right?

Reply:
yes, correct. it was doing ncols so getting an extra colgap in the listings case (whcih doesn't have row labels so needs ncols - 1 colgaps for ncols columns. tables have row labels so they need ncols colgaps

Reply:
Super! That was the source of the error then. Thanks for the fix ;)